arangod/RocksDBEngine/RocksDBIncrementalSync.cpp

```diff
@@ -505,6 +505,8 @@ Result handleSyncKeysRocksDB(DatabaseInitialSyncer& syncer,
 
     trx.addHint(transaction::Hints::Hint::RECOVERY);  // turn off waitForSync!
     trx.addHint(transaction::Hints::Hint::NO_INDEXING);
+    // turn on intermediate commits as the number of operations can be huge here
+    trx.addHint(transaction::Hints::Hint::INTERMEDIATE_COMMITS);
```

Comment:
shouldn't we add the same flag a few lines up (around line 460-470) as well?
I think it is missing there, and the removal operation could also include a lot of documents.